### ISSUE TITLE
Event handlers duplicate after each auto-reload

### ISSUE THREAD

**Issue by andreemic** [gradio-app/gradio#5789]
### Describe the bug

At every source-file change and auto-reload of the app, simple event handlers (e.g. attached to "click" event of buttons) duplicate and get executed multiple times when triggered.

So at first: 1 click -> 1 execution.
After one hot-reload: 1 click -> 2 executions
After two hot-reloads: 1 click -> 3 executions
After n hot-reloads: 1 click -> n executions.

This makes hot-reload unusable with any application that has side effects or uses the GPU in the background because it gets overloaded by the multiple parallel executions. Which are most applications of Gradio as far as I understand.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

1. Create a file "app.py" with the following contents:
```python
import gradio as gr

def greet(name):
    print(name)
    
    return "Hello " + name + "!!"

demo = gr.Interface(fn=greet, inputs="text", outputs="text")
    
demo.launch()   
```

2. launch it with `gradio app.py`
3. Try changing the code to trigger an auto-reload (inserting a line) and clicking "submit" within the Gradio app. Watch the console. You will see the handler being executed more and more after each hot-reload:
 
```
$ gradio app.py 
Watching: '...'

Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
before first hot-reload
Changes detected in: app.py
after one hot-reload
after one hot-reload
Changes detected in: app.py
after two hot-reloads
after two hot-reloads
after two hot-reloads
Changes detected in: app.py
after three hot-reloads
after three hot-reloads
after three hot-reloads
after three hot-reloads
```

### Screenshot

<img width="827" alt="Screenshot 2023-10-04 at 13 47 30" src="https://github.com/gradio-app/gradio/assets/27695722/58b53db2-036e-4925-8690-74a36863a16a">

### Logs

```shell
$ gradio app.py 
Watching: '...'

Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
before first hot-reload
Changes detected in: app.py
after one hot-reload
after one hot-reload
Changes detected in: app.py
after two hot-reloads
after two hot-reloads
after two hot-reloads
Changes detected in: app.py
after three hot-reloads
after three hot-reloads
after three hot-reloads
after three hot-reloads
```

### System Info

```shell
Gradio Environment Information:
------------------------------
Operating System: Linux
gradio version: 3.43.2
gradio_client version: 0.5.0

------------------------------------------------
gradio dependencies in your environment:

aiofiles: 23.2.1
altair: 5.1.1
fastapi: 0.103.1
ffmpy: 0.3.1
gradio-client==0.5.0 is not installed.
httpx: 0.25.0
huggingface-hub: 0.17.2
importlib-resources: 6.0.1
jinja2: 3.1.2
markupsafe: 2.1.2
matplotlib: 3.8.0
numpy: 1.25.2
orjson: 3.9.7
packaging: 23.1
pandas: 2.1.0
pillow: 10.0.1
pydantic: 2.3.0
pydub: 0.25.1
python-multipart: 0.0.6
pyyaml: 6.0
requests: 2.29.0
semantic-version: 2.10.0
typing-extensions: 4.7.1
uvicorn: 0.23.2
websockets: 11.0.3
authlib; extra == 'oauth' is not installed.
itsdangerous; extra == 'oauth' is not installed.


gradio_client dependencies in your environment:

fsspec: 2023.9.1
httpx: 0.25.0
huggingface-hub: 0.17.2
packaging: 23.1
requests: 2.29.0
typing-extensions: 4.7.1
websockets: 11.0.3
```

### Severity

Blocking usage of gradio

**Comment by pngwn** [comment]
Oh. I might know what this is. 

Thanks for the report @andreemic!

**Comment by andreemic** [comment]
Any more ways I can help?

**Comment by dmahlow** [comment]
I'm seeing this as well with version 4.4.1. What's the root cause, is there a workaround?

**Comment by ziontee113** [comment]
I'm running into this issue as well. It makes hot-reload unusable.

**Comment by abidlabs** [comment]
This is fixed now, as of `gradio== 4.14.0`, thanks to @aliabid94. Sorry for the delay!
